src/Build/BackEnd/Node/RarNode.cs

```diff
@@ -38,11 +37,12 @@ public sealed class RarNode : INode
         public NodeEngineShutdownReason Run(bool nodeReuse, bool lowPriority, out Exception shutdownException, CancellationToken cancellationToken = default)
         {
             shutdownException = null;
-            using CancellationTokenSource cts = new CancellationTokenSource();
+            using CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
             string pipeName = CommunicationsUtilities.GetRarPipeName(nodeReuse, lowPriority);
             Handshake handshake = NodeProviderOutOfProc.GetHandshake(enableNodeReuse: nodeReuse,
                                                                      enableLowPriority: lowPriority, specialNode: true);
 
+            Console.WriteLine(pipeName);
```

Comment:
Can you switch this to use the MSBUILDDEBUGCOMM infrastructure (like `CommunicationsUtilities.Trace`) instead?

Reply:
Forgot to delete this, but after some thoughts this could be helpful to have in some logs. So I have used the `CommunicationsUtilities.Trace` instead.